source/common/http/conn_manager_impl.cc

```diff
@@ -366,6 +367,7 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
   new_stream->response_encoder_->getStream().addCallbacks(*new_stream);
   new_stream->response_encoder_->getStream().setFlushTimeout(new_stream->idle_timeout_ms_);
   new_stream->streamInfo().setDownstreamBytesMeter(response_encoder.getStream().bytesMeter());
+  new_stream->streamInfo().setStreamState(StreamInfo::StreamState::InProgress);
```

Comment:
I think that this might be too early for InProgress. We haven't yet even received complete headers.

Reply:
Updated it, let me know if this makes more sense.